Tangem/Modules/Referral/ReferralView.swift

```diff
@@ -12,63 +12,195 @@ struct ReferralView: View {
-            
-            ReferralPlaceholderPointView { Assets.discount }
+            ReferralPlaceholderPointView(icon: Assets.cryptoCurrencies)
+
+            ReferralPlaceholderPointView(icon: Assets.discount)
         }
         .padding(.horizontal, 16)
     }
-}
 
+    @ViewBuilder
+    var touButton: some View {
+        Button(action: viewModel.openTou) {
+            Text(viewModel.touButtonPrefix) +
+                Text("common_terms_and_conditions").foregroundColor(Colors.Text.accent) +
+                Text("referral_tou_suffix")
+        }
+        .multilineTextAlignment(.center)
+        .fixedSize(horizontal: false, vertical: true)
+        .font(Fonts.Regular.footnote)
+        .foregroundColor(Colors.Text.tertiary)
+        .padding(.horizontal, 20)
+    }
+
+    @ViewBuilder
+    var alreadyReferralBottomView: some View {
+        VStack(spacing: 14) {
+            Spacer()
+
+            HStack {
+                Text("referral_friends_bought_title")
+
+                Spacer()
+
+                Text(viewModel.numberOfWalletsBought)
+            }
+            .padding(16)
+            .background(Colors.Background.primary)
+            .cornerRadius(14)
+            .padding(.top, 24)
 
+            VStack(spacing: 8) {
+                Text("referral_promo_code_title")
+                    .font(Fonts.Bold.footnote)
+                    .foregroundColor(Colors.Text.tertiary)
+
+                Text(viewModel.promoCode)
+                    .font(Fonts.Regular.title1)
+                    .foregroundColor(Colors.Text.primary1)
+                    .fixedSize(horizontal: false, vertical: true)
+            }
+            .frame(maxWidth: .infinity)
+            .padding(.top, 8)
+            .padding(.bottom, 12)
+            .background(Colors.Background.primary)
+            .cornerRadius(14)
+
+            HStack(spacing: 12) {
+                TangemButton(title: "common_copy",
+                             systemImage: "square.on.square",
+                             iconPosition: .leading,
+                             iconPadding: 10,
+                             action: viewModel.copyPromoCode)
+                    .buttonStyle(TangemButtonStyle(colorStyle: .black,
+                                                   layout: .flexibleWidth))
+
+                TangemButton(title: "common_share",
+                             systemImage: "arrowshape.turn.up.forward",
+                             iconPosition: .leading,
+                             iconPadding: 10,
+                             action: viewModel.sharePromoCode)
+                    .buttonStyle(TangemButtonStyle(colorStyle: .black,
+                                                   layout: .flexibleWidth))
+            }
+
+            touButton
+        }
+    }
+
+    @ViewBuilder
+    var notReferralView: some View {
+        VStack(spacing: 0) {
+            touButton
+                .padding(.bottom, 12)
```

Comment:
```suggestion
        VStack(spacing: 12) {
            touButton
```
может лучше через `spacing`?

Reply:
да, ок